src/Rules/DecoupledCode/NoHelperDependingOnBootedApplicationUsageRule.php

```diff
@@ -0,0 +1,96 @@
+{
+    private const GLOBAL_HELPERS_ALTERNATIVES = [
+        'app_path'      => 'Inject Illuminate\Contracts\Foundation\Application and use method "path"',
+        'base_path'     => 'Inject Illuminate\Contracts\Foundation\Application and use method "basePath"',
+        'config_path'   => 'Inject Illuminate\Contracts\Foundation\Application and use method "configPath"',
+        'database_path' => 'Inject Illuminate\Contracts\Foundation\Application and use method "databasePath"',
+        'resource_path' => 'Inject Illuminate\Contracts\Foundation\Application and use method "resourcePath"',
+        'public_path'   => 'Inject Illuminate\Contracts\Foundation\Application and use method "publicPath"',
+        'lang_path'     => 'Inject Illuminate\Contracts\Foundation\Application and use method "langPath"',
+        'storage_path'  => 'Inject Illuminate\Contracts\Foundation\Application and use method "storagePath"',
+        'resolve'       => 'Inject Illuminate\Contracts\Foundation\Application and use method "make"',
+        'app'           => 'Inject Illuminate\Contracts\Foundation\Application and use method "make"',
+        'abort'         => 'Inject Illuminate\Contracts\Foundation\Application and use method "abort"',
+        'abort_if'      => 'Inject Illuminate\Contracts\Foundation\Application and use method "abort" within an if statement',
+        'abort_unless'  => 'Inject Illuminate\Contracts\Foundation\Application and use method "abort" within an if statement',
+        '__'            => 'Inject Illuminate\Translation\Translator and use method "translate"',
+        'trans'         => 'Inject Illuminate\Translation\Translator and use method "translate"',
+        'trans_choice'  => 'Inject Illuminate\Translation\Translator and use method "choice"',
+        'action'        => 'Inject Illuminate\Contracts\Routing\UrlGenerator and use method "action"',
+        'asset'         => 'Inject Illuminate\Contracts\Routing\UrlGenerator and use method "asset"',
+        'secure_asset'  => 'Inject Illuminate\Contracts\Routing\UrlGenerator and use method "asset" with the second parameter set to "true"',
+        'route'         => 'Inject Illuminate\Contracts\Routing\UrlGenerator and use method "route"',
+        'url'           => 'Inject Illuminate\Contracts\Routing\UrlGenerator and use method "url", "current" for the current url, "full" for the full url or "previous" for the previous url',
+        'secure_url'    => 'Inject Illuminate\Contracts\Routing\UrlGenerator and use method "url" with the third parameter set to "true"',
+        'redirect'      => 'Inject Illuminate\Routing\Redirector and use method "to"',
+        'to_route'      => 'Inject Illuminate\Routing\Redirector and use method "route"',
+        'back'          => 'Inject Illuminate\Routing\Redirector and use method "back"',
+        'config'        => 'Inject Illuminate\Config\Repository and use method "all" or "get"',
```

Comment:
This rule is genius 


However, would this be better to recommend the config repository contract? As that would provide the most decoupling whilst working the same

To be honest nearly all of these cases have contracts that could be suggested rather than concrete classes, except for the managers perhaps 

I'm away from my pc atm so I can't research them all but that would provide the most decoupling and state independency

Reply:
Good point! Will update them to use the contracts as much as possible!

Reply:
I moved allmost all of the suggestion to contracts/interfaces, the only ones I didn't find a contract for where the Request, the Redirector and the SessionManager, as those are standalone classes.